nhueck.GroupTasks/src/groupTasks.js

```diff
@@ -0,0 +1,241 @@
+// groups all open tasks of the current note according to their first #tag
+// and adds them below a heading "Tag"
+// Author: Nikolaus Hueck twitter: @nikolaushueck
+//------------------------------------------------------------------------------
+//------------------------------------------------------------------------------
+
+import { getOrMakeConfigurationSection } from '../../nmn.Templates/src/configuration'
+
+const DEFAULT_GROUPTASKS_OPTIONS = `  groupTasks: {
+    allTasksHeading: "Tasks",
+    noHashtagName: "NoTag",
+    deleteEmptyLines: true,
+    taggedTasksHeadingsLevel: 4,
+    deleteEmptyTasksHeadings: true,
+  },
+`
+let groupTasksConfig = {
+  allTasksHeading : "Tasks", 
+  noHashtagName: "NoTag",
+  deleteEmptyLines : true, 
+  taggedTasksHeadingsLevel: 4, 
+  deleteEmptyTasksHeadings: true
+}
+
+//---------------------------------------
+// helper function: getPriority(string)
+// get the priority og a given task
+// returns a number (1 - 3) 3 beeing the highest priority
+//---------------------------------------
+function getPriority(s) {
+  
+}
+
+
+//---------------------------------------
+// helper function: lineSpaceHeadings()
+// 1. delete multiple empty lines above headings
+// 2. delete headings of level 4 that aren't followed by text paragraphs
+//---------------------------------------
+
+// eslint-disable-next-line max-len
+function lineSpaceHeadings(level, deleteEmptyLines, deleteEmptyTasksHeadings) {
+  // console.log("lineSpaceHeadings wird ausgeführt")
+  
+  if(deleteEmptyLines) { 
+    console.log("delete empty lines")
+    Editor.content = Editor.content.replace(/(\n)+#/g, "\n#")  // delete multiple empty lines above headings
+  } 
+  if(deleteEmptyTasksHeadings) {  
+    const re = "(\n)+"+"#".repeat(level)+" .+(?=\n*?((# )|(##)))"
+    const re1 = "(\n)+"+"#".repeat(level)+" .+(\n)*$"  //$
+    console.log("deleting last line: "+re)
+    const taskHeadingSearch = new RegExp(re,"g")
+    const taskHeadingSearch1 = new RegExp(re1,"g")
+    Editor.content = Editor.content.replace(taskHeadingSearch, "")  // delete headings of the tasksHeadings level that aren't followed by any text
+    Editor.content = Editor.content.replace(taskHeadingSearch1, "")
+  }
+}
+
+//---------------------------------------
+// helper function: capitalizeFirstLetter(s)
+// Capitalize the first letter of a string
+//---------------------------------------
+
+function capitalizeFirstLetter(s) {
+  return s.charAt(0).toUpperCase() + s.slice(1)
+}
+
+//---------------------------------------
+// helper function: stripHashFromHashtag(s)
+// Deletes all "#"s from a string
+//---------------------------------------
+
+function stripHashFromHashtag(s) {
+  return s.replace('#', '')
+}
+
+//---------------------------------------
+// helper function: stripAsteriskFromTodo(s)
+// Deletes all "*{space}"s in a string
+//---------------------------------------
+
+function stripAsteriskFromTodo(s) {
+  return s.replace('* ', '')
```

Comment:
I don't think this is a good way of doing this, as
a) there are at least two other ways of marking tasks.
I suggest you instead look at the type ParagraphType ("open" | "done" | "scheduled" | "cancelled" | "title" | "quote" | "list" | "empty" | "text") to look for "open" task types.